File: frontend/Shelter/src/app/core/models/animal-query.ts
import { AnimalStatus, Sex } from "./animal";

export type SortBy = 'name' | 'species' | 'birthDate' | 'sex' | 'status' | 'createdAt';
export type SortDir = 'asc' | 'desc';

export interface AnimalQuery {
  // sorting
  sortBy?: SortBy;
  sortDir?: SortDir;

  // pagination
  page?: number;
  size?: number;

  // filtering
  speciesId?: number | null;
  sex?: Sex;
  status?: AnimalStatus;

  // age in months
  ageMinMonths?: number;
  ageMaxMonths?: number;

  // created range (ISO date strings)
  createdFrom?: string; // 'yyyy-MM-dd'
  createdTo?: string;   // 'yyyy-MM-dd'

  vaccinated?: boolean;
  neutered?: boolean;  
}

File: backend/Shelter/Application/Queries/AnimalQuery.cs
using Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Application.Queries
{
    public class AnimalQuery
    {
        //sorting
        public AnimalSortField SortBy { get; set; } = AnimalSortField.createdAt;
        public SortDirection SortDir { get; set; } = SortDirection.desc;

        //pagination
        public int Page { get; set; } = 0;
        public int Size { get; set; } = 20;

        //filters
        public int? SpeciesId { get; set; }
        public Sex? Sex { get; set; }
        public AnimalStatus? Status { get; set; }

        //age in months
        public int? AgeMinMonths { get; set; }         // np. 6
        public int? AgeMaxMonths { get; set; }         // np. 120

        // zakres createdAt
        public DateTime? CreatedFrom { get; set; }
        public DateTime? CreatedTo { get; set; }

        // flags
        public bool? Vaccinated { get; set; }
        public bool? Neutered { get; set; }


    }
}
